### ISSUE TITLE
Closed loop VVT

### ISSUE THREAD

**Issue by ByteVenom** [noisymime/speeduino#175]
Currently there isn't precise control available for VVT. Duty cycle on the 01-05 Miata VVT solenoid doesn't translate to the same VVT angles always. Closed loop control of VVT for the 99-05 Miata trigger pattern would be very helpful, especially if it could be expanded to other trigger patterns.

**Comment by mattseabrook** [comment]
Definitely agree this would be a valuable addition, especially for engines like the Miata where duty cycle ≠ consistent angle.

A minimal closed-loop approach could use the existing cam vs crank position to determine actual angle and apply a PID loop to adjust the solenoid's PWM:

   ```c
    int error = target_angle - cam_angle;
    integral += error;
    duty += (kp * error) + (ki * integral);
    duty = constrain(duty, 0, 100);
```
Target angle could come from a simple RPM vs. MAP or TPS table. Cam angle phase is already derivable from existing triggers.

If maintainers are open to it, I can prototype this as `calculateVVT()` using existing scheduler structure.

**Comment by adbancroft** [comment]
Isn't this now available?

**Comment by theKraid** [comment]
Yep I got closed loop VVT running and working very well on a miata. According to the wiki, missing tooth and some other decoder also support it.
